stl/inc/expected

```diff
@@ -252,9 +256,10 @@ public:
                                            && !is_constructible_v<unexpected<_Err>, const expected<_Uty, _UErr>&> //
                                            && !is_constructible_v<unexpected<_Err>, const expected<_Uty, _UErr>>;
 
+public:
     template <class _Uty, class _UErr>
-        requires is_constructible_v<_Ty, const _Uty&> && is_constructible_v<_Err, const _UErr&>
-              && _Allow_unwrapping<_Uty, _UErr>
+        requires _Not_self<_Uty, _UErr> && is_constructible_v<_Ty, const _Uty&>

```

Comment:
We conventionally use `_Different_from` (which is the Standard's exposition-only concept _`different-from`_) for this. Also, thanks for adding more fuel to my "every converting constructor needs to be constrained to not act like a special member function" rule of thumb ;)